### ISSUE TITLE
How to train SVM model by Python?

### ISSUE THREAD

**Issue by LunaLuan** [pillarpond/face-recognizer-android#4]
Hi there. I want to use your model for my management application. I want to train SVM in server-side and push model to Android application, but the results is very poor.  Can you explain the process to training data?

**Comment by LunaLuan** [comment]
This is my process:
### Likely your file: Facenet.java
#### For image standardize: standardized_images = normalize_meanstd(rgb_images, axis=(1,2))
def normalize_meanstd(a, axis=None): 
    # axis param denotes axes along which mean & std reductions are to be performed
    mean = np.mean(a, axis=axis, keepdims=True)
    std = np.sqrt(((a - mean)**2).mean(axis=axis, keepdims=True))
    
    return (a - mean) / std 


detect = face.Detection()

### I use your freeze model.
encoder = face.Encoder('facenet.pb')

embedded_size = 512
embedded = np.zeros((metadata.shape[0], embedded_size))
targets = np.array([m.name for m in metadata])

for i, m in enumerate(metadata):
    img = load_image(m.image_path())
    faces = detect.find_faces(img)
    
    temp_face = face.Face()
    temp_face.image = normalize_meanstd(faces[0].image, axis=(1,2))
    
    embedded[i] = encoder.generate_embedding(temp_face)


from sklearn.preprocessing import LabelEncoder

label_encoder = LabelEncoder()
label_encoder.fit(targets)

y = label_encoder.transform(targets)

train_idx = np.arange(metadata.shape[0]) % 2 == 0
test_idx = np.arange(metadata.shape[0]) % 2 != 0

X_train = embedded[train_idx]
X_test = embedded[test_idx]

y_train = y[train_idx]
y_test = y[test_idx]


### Training with linear inference...
m = svm_train(y_train.tolist(), X_train.tolist(), '-t 0 -b 1')
p_label, p_acc, p_val = svm_predict(y_test.tolist(), X_test.tolist(), m, '-b 1')


# In[20]:


svm_save_model(model_save_path + '/model', m)

file = open(model_save_path + '/label', 'w')
for i in label_encoder.classes_:
    file.write(i + '\n')
file.close()

**Comment by LunaLuan** [comment]
> Hi there. I want to use your model for my management application. I want to train SVM in server-side and push model to Android application, but the results is very poor. Can you explain the process to training data?

**Comment by LunaLuan** [comment]
My process: https://gist.github.com/LunaLuan/ffa31373583e09832ff617600b74a170

**Comment by pillarpond** [comment]
How many pictures were used to training model? 
I've used pictures at least 15 for each person.

And, I think it is better that you refer to davidsandberg's facenet.
https://github.com/davidsandberg/facenet/blob/096ed770f163957c1e56efa7feeb194773920f6e/src/classifier.py#L92

**Comment by LunaLuan** [comment]
Thank you very much.

But I resuse davidsandberg's code, I use your pb files with davidsandberg's facenet. Example:
Embbeding vector by Android Application:
'10 0:0.0018329972 1:-0.01819302 2:-0.009794952 3:0.024219211 4:0.027110105 5:0.011321024 .... '

Embedding vector by davidsandberg's code:
array([-3.72377364e-03,  1.27700809e-02,  4.67353780e-03,  8.95703211e-02,
        5.98715208e-02, -1.10814571e-02, -1.64364316e-02, -5.27036190e-03,...

I think your processing and davidsandberg's code are different.

**Comment by pillarpond** [comment]
You can make data file using below code

```python
with open('data', 'w') as outfile:
    for i in range(len(labels)):
        outfile.write('%d' % labels[i])
        for j in range(len(emb_array[i])):
            outfile.write(' %d:%f' % (j, emb_array[i][j]))
        outfile.write('\n')
```
